### PR TITLE
Date picker component

### LINKED ISSUE #46: enhanced time range picker component
Currently the month picker component only allows users to select ranges of data by months. In most cases, users are concerned with the last week's set of data. In order to provide further granularity to time range selection, we will build a component to allow users to set relative and absolute time ranges. 

Example design: https://dribbble.com/shots/4433950-Datepicker-with-Time-Range-Slider

### PR DESCRIPTION
**Pull request by aquibbaig**
I'll remove the previous PR #61 and will be tracking up the progress of date-picker-component in this PR. Fixes #46.

This was a large PR cause it modifies existing endpoints we had and the test suites as well.

### CODE REVIEW

**Comment by gurbirkalsi** [comment]
@aquibbaig Converting this PR to a draft for now as we work on implementing the Patternfly CSS spec Mary provided us.

**Comment by aquibbaig** [comment]
@gurbirkalsi I think `RangePicker` component from antd doesn't offer a styling which is too close to the current pf implementation. I think we can use two `DatePicker` components together and use a wrapper around it to achieve this feature.

**Comment by aquibbaig** [comment]
Screenshot 2020-06-30 at 1 42 28 PM

@gurbirkalsi @portante As per Mary's design implementations

File: src/components/DatePicker/index.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  selectedDateRange: global.selectedDateRange,
+}))
+class AntdDatePicker extends Component {
+  handleChange = (dates, dateStrings) => {
+    const { dispatch, onChangeCallback } = this.props;
+    dispatch({
+      type: 'global/updateSelectedDateRange',
+      payload: dateStrings,
+    }).then(() => {
+      if (onChangeCallback != null) {
```
Comment (gurbirkalsi) [inline src/components/DatePicker/index.js]: What if we made `onChangeCallback` a required prop as opposed to checking whether it exists before invocation?

File: src/components/DatePicker/index.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import styles from './index.less';
+
+const { RangePicker } = DatePicker;
+
+@connect(({ global, datastore }) => ({
+  indices: datastore.indices,
+  selectedDateRange: global.selectedDateRange,
+}))
+class AntdDatePicker extends Component {
+  handleChange = (dates, dateStrings) => {
```
Comment (gurbirkalsi) [inline src/components/DatePicker/index.js:23]: `dates` isn't being referenced in `handleChange`, do we still need this param here?

File: src/pages/ComparisonSelect/index.js [hunk 3]
```diff
@@ -98,6 +98,7 @@ class ComparisonSelect extends React.Component {
       <PageHeaderLayout
         title={selectedControllers.join(', ')}
         selectedControllers={selectedControllers}
+        key="pageheader"
```
Comment (gurbirkalsi) [inline src/pages/ComparisonSelect/index.js]: Why do we need to specify a key for `PageHeaderLayout` here?

File: src/models/global.js [hunk 4]
```diff
@@ -8,7 +8,7 @@ export default {
     sessionDescription: '',
     sessionId: '',
     collapsed: true,
-    selectedIndices: [],
+    selectedDateRange: [],
```
Comment (gurbirkalsi) [inline src/models/global.js]: Can we consider making `selectedDateRange` an object with `start` and `end` keys for readability purposes throughout the code?

File: src/components/DatePicker/index.js (hunk 2 above)
Comment (aquibbaig) [reply src/components/DatePicker/index.js:23]: @gurbirkalsi Actually, when the user selects a date from the datepicker, there are two values returned by the event: 
- `dates`: which are moment objects
- `dateStrings`: which are moment formatted strings.

If I just reference one parameter here, then it automatically behaves as the `dates` in the default parameter

Comment (gurbirkalsi) [reply src/components/DatePicker/index.js:23]: Thanks for the clarification!

File: src/models/dashboard.js [hunk 5]
```diff
@@ -37,26 +37,30 @@ export default {
...
-          lastModStr = controller.runs_preV1.value_as_string;
-        }
-        controllers.push({
-          key: controller.key,
-          controller: controller.key,
-          results: controller.doc_count,
-          last_modified_value: lastModVal,
-          last_modified_string: lastModStr,
+      // When there are no hits for specific indices,
+      // there are no aggregations returned.
```
Comment (aquibbaig) [inline src/models/dashboard.js]: Remove the comment

File: src/pages/Search/index.js [hunk 6]
```diff
@@ -34,21 +34,25 @@ class SearchList extends Component {
...
 
   componentDidMount() {
-    const { indices, mapping, selectedIndices } = this.props;
+    const { indices, mapping, selectedDateRange } = this.props;
 
-    if (indices.length === 0 || Object.keys(mapping).length === 0 || selectedIndices.length === 0) {
+    if (
+      indices.length === 0 ||
+      Object.keys(mapping).length === 0 ||
+      selectedDateRange.length === 0
```
Comment (aquibbaig) [inline src/pages/Search/index.js]: Check if it's an empty object

**Comment by gurbirkalsi** [comment]
@aquibbaig Great work! Can you please squash these commits down to one before we merge this work?

**Comment by aquibbaig** [comment]
@gurbirkalsi Done! :sparkles:

**Pull request merged**